### PR TITLE
Changes to tests.

### PR DESCRIPTION
**Pull request by Insti**
These changes make all the tests pass on the linewise-formatter branch.
Split up into the 3 things that need to change.

### CODE REVIEW

**Comment by Insti** [comment]
I suggest you merge them WITHOUT squashing.

**Comment by k3rni** [comment]
Thanks! Was just about to do similar changes. One thing is that we probably can't use the swift example as-is, like you've pointed out [here](/exercism/exercism.io/pull/3058#discussion_r74927481). I'll merge and replace with a slimmer, contrived example.

File: test/exercism/html_formatter_test.rb [hunk 1]
```diff
@@ -10,14 +10,15 @@ def setup
 
   def test_ocaml_comment
     tokens = load_sample 'ocaml', 'ocaml'
-    doc = parse @formatter.format(tokens)
+    html = @formatter.format(tokens)
+    doc = parse html
     raw_lines = load_example('ocaml').lines
 
-    assert_equal 6, doc.css('div[id^=L]').size, "Formatting should preserve linecount"
+    assert_equal 6, doc.css('span[id^=L]').size, "Formatting should preserve linecount"
```
Comment (Insti) [inline test/exercism/html_formatter_test.rb:17]: Should these be `span`s or `div`s?

Comment (k3rni) [inline test/exercism/html_formatter_test.rb:17]: Spans, as we're aiming for drop-in compatibility. Using divs and removing the extra newline in front of closing tags may be a future matter.

Comment (Insti) [inline test/exercism/html_formatter_test.rb:17]: I checked the rendering in a browser and it looks like `span`s work.

**Pull request merged**